src/bootstrap/src/core/config/config.rs

```diff
@@ -2484,3 +2480,15 @@ fn find_correct_section_for_field(field_name: &str) -> Vec<WouldBeValidFor> {
         })
         .collect()
 }
+
+fn test_build_dir() -> PathBuf {
```

Comment:
Could you please add a comment here that explains when is `CARGO_TARGET_DIR` set (when we execute tests through bootstrap) and when not (when we run them with `cargo test`)?